src/ReactiveSocket.cpp

```diff
@@ -35,14 +35,15 @@ ReactiveSocket::ReactiveSocket(
     bool isServer,
     std::shared_ptr<RequestHandlerBase> handler,
     Stats& stats,
-    std::unique_ptr<KeepaliveTimer> keepaliveTimer)
+    std::unique_ptr<KeepaliveTimer> keepaliveTimer,
+    folly::Executor& executor)
     : handler_(handler),
       connection_(std::make_shared<ConnectionAutomaton>(
-          [handler](
+          [this, handler](
```

Comment:
Is it guaranteed that the `ConnectionAutomaton` will be destroyed before the `ReactiveSocket`? Since they're both `shared_ptr`s, it seems possible that the automaton could outlive the socket in some pathological cases.